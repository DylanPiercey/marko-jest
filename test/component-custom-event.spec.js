const path = require('path');
const { initComponent, createTestSandbox } = require('../test-utils');

describe('component with custom event', () => {
  const componentClass = initComponent(path.resolve(__dirname, './resources/list-item/index.marko'));
  const MODEL = { label: 'one', value: 1 };

  let testSandbox;
  let component;

  beforeEach(() => {
    testSandbox = createTestSandbox();
  });

  afterEach(() => {
    testSandbox.reset();
  });

  describe('on rendering', () => {
    beforeEach(async () => {
      component = await testSandbox.renderComponent(componentClass, MODEL);
    });

    it('should render correctly', () => {
      expect(component.els).toMatchSnapshot();
    });

    describe('when clicked', () => {
      let onSelect;

      beforeEach(async () => {
        onSelect = jest.fn();
        component.on('select', onSelect);

        component.el.click();
      });

      afterEach(() => {
        component.removeListener('select', onSelect);
      });

      it('should emit custom event with the right parameter', () => {
        expect(onSelect).toHaveBeenCalledWith(1);
      });
    });
  });
});
