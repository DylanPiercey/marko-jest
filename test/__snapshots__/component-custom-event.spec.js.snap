// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component with custom event on rendering should render correctly 1`] = `
Array [
  ,
  <li>
    one
  </li>,
  ,
]
`;
